tests/std/tests/GH_002885_stable_sort_difference_type/test.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <cstddef>
+#include <cstdint>
+#include <ranges>
+
+#include "range_algorithm_support.hpp"
+
+using namespace std;
+
+constexpr auto pred = [](int i) { return i <= 42; };
+
+template <class I>
+void test_iota_transform() {
+    constexpr int orig[]{42, 1729};
+    int a[]{42, 1729};
+    auto vw = views::iota(I{}, static_cast<I>(ranges::size(a)))
+            | views::transform([&a](I i) -> auto& { return a[static_cast<size_t>(i)]; });
+
+    static_assert(three_way_comparable<ranges::iterator_t<ranges::iota_view<I>>>); // TRANSITION, /permissive

```

Comment:
This is really weird. It seems that MSVC doesn't evaluate concept specializations as quickly as required in permissive modes.

Reply:
From the last bit of the error log:
```
D:\build\out\inc\__msvc_ranges_to.hpp(865): note: while evaluating concept 'three_way_comparable<std::ranges::_Ioterator<signed char>,std::partial_ordering>'
D:\build\out\inc\__msvc_ranges_to.hpp(865): error C2131: expression did not evaluate to a constant
D:\build\out\inc\__msvc_ranges_to.hpp(865): note: failure was caused by a read of an uninitialized symbol
D:\build\out\inc\__msvc_ranges_to.hpp(865): note: see usage of 'std::three_way_comparable<std::ranges::_Ioterator<_Wi>,std::partial_ordering>'
```
"C2131 while evaluating `meow` ... see usage of `meow`" generally suggests constraint recursion, which is not that unusual in permissive mode. The fact the overload resolution sees "extra" friend overloads that ought to be hidden can easily trigger constraint recursion.